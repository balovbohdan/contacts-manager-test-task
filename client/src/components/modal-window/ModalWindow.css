.main {
    z-index: 99999;
    position: absolute;

    width: 100vw;
    height: 100vh;

    top: 0;
    left: 0;
    right: 0;
    bottom: 0;

    display: flex;
    align-items: center;
    justify-content: center;
}

.body {
    max-width: 450px;
    width: calc(100vw - 20px);

    min-height: 200px;
    max-height: calc(100vh - 40px);

    padding: 10px;
    margin: 20px auto;

    position: relative;
    border-radius: var(--border-radius);
    background-color: var(--color-light-blue);
    box-shadow: 0 20px 35px rgba(0, 0, 0, .3), 0 15px 12px rgba(0, 0, 0, .2);

    color: var(--color-white);
}

.head {
    position: relative;
    padding-bottom: 10px;

    display: flex;
    align-items: center;
    justify-content: flex-start;
}

.closeBtn {
    position: absolute;

    top: -20px;
    right: -20px;

    width: 40px;
    height: 40px;

    display: flex;
    align-items: center;
    justify-content: center;

    cursor: pointer;
    border-radius: 50%;
    transition: var(--trans);
    background-color: var(--color-red);
    box-shadow: 0 3px 10px rgba(0, 0, 0, .3), 0 5px 10px rgba(0, 0, 0,.2);
}

.closeBtn:hover {
    transform: rotate(10deg) scale(.9);
}

.closeBtn_img {
    width: calc(100% - 20px);
    height: calc(100% - 20px);

    background-size: contain;
    background-repeat: no-repeat;
    background-position: center center;
    background-image: url(~@img/close.svg);
}