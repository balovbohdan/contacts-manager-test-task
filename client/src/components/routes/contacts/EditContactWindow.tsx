import * as React from 'react';

import {Contacts} from '@lib/entities/contacts/contact/types';
import {EditContact} from '@storehouse/actions/contacts/types';

import {
    EditContactWindow as Base
} from '@components/contacts-manager/edit-contact-window';

type Props = {
    close:()=>void;
    edit:EditContact;

    contacts:Contacts;
    active:boolean|number;
};

export const EditContactWindow = (props:Props) => {
    const active = +props.active || false;

    if (typeof active !== 'number')
        return null;

    const contact = props.contacts[active];

    return typeof contact !== 'object'
        ? null
        : <Base
            contact={contact}
            edit={props.edit}
            close={props.close}/>;
};