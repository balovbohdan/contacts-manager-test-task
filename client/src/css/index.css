@import url(reset.css);
@import url(fonts.css);
@import url(titles.css);
@import url(constants.css);
@import url(main.css);