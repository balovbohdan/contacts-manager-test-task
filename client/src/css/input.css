.label {
    font-size: .9rem;
}

.input {
    width: 100%;
    padding: 5px 0;
    border-bottom: 2px solid var(--color-yellow);

    font-size: 1.5rem;
    color: var(--color-white);
}

@media (max-width: 600px) {
    .input {
        font-size: 1.2rem;
    }

    .label {
        font-size: .8rem;
    }
}

@media (max-width: 400px) {
    .input {
        font-size: 1.1rem;
    }

    .label {
        font-size: .7rem;
    }
}