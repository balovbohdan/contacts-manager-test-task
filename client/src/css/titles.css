h1, h2, h3, h4, h5, h6 {
    font-size: 100%;
    font-family: var(--font-main);
}

h1 {
    font-size: 40px;
}

h2 {
    font-size: 35px;
}

h3 {
    font-size: 30px;
}

h4 {
    font-size: 25px;
}

h5 {
    font-size: 20px;
}

h6 {
    font-size: 17px;
}

@media (max-width: 550px) {
    h1 {
        font-size: 35px;
    }

    h3 {
        font-size: 25px;
    }
}

@media (max-width: 400px) {
    h1 {
        font-size: 30px;
    }

    h3 {
        font-size: 20px;
    }
}