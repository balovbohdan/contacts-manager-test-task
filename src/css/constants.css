:root {
    --color-red: #FB4B4E;
    --color-blue: #16324F;
    --color-white: #F3F7F0;
    --color-yellow: #FDE74C;
    --color-light-blue: #236098;

    --font-main: Roboto, serif;
    --font-marker: 'Permanent Marker', serif;

    --border-radius: 3px;
    --header-height: 60px;
    --trans: all .12s ease-in;
}