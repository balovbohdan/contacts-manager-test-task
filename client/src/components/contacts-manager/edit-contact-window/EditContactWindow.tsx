import * as React from 'react';

import {ModalWindow} from '@components/modal-window';

import {Props} from './types';
import {createSubmit} from './utils';
import {ContactForm} from '../contact-form';

export const EditContactWindow = ({edit, close, contact}:Props) => {
    const {id, tip, name, phone} = contact;
    const title = `Edit «${name}»`;
    const def = { tip, name, phone };
    const submit = createSubmit(id, close, edit);

    return (
        <ModalWindow close={close} title={title}>
            <ContactForm def={def} submit={submit}/>
        </ModalWindow>
    );
};