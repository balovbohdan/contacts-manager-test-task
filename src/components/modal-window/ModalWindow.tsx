import * as React from 'react';

import {Fader} from '@components/fader';

import css from './ModalWindow.css';

type Props = {
    close:Close;
    title:string;
    children:React.ReactNode;
};

type HeadProps = {
    close:Close;
    title:string;
};

type CloseBtnProps = {
    close:Close;
};

type Close = ()=>void;

export const ModalWindow = ({close, title, children}:Props) => {
    const onWrapperClick = createOnWrapperClick(close);

    return (
        <Fader>
            <div className={css.main} onClick={onWrapperClick}>
                <div className={css.body}>
                    <Head close={close} title={title}/>
                    {children}
                </div>
            </div>
        </Fader>
    );
};

const Head = ({close, title}:HeadProps) =>
    <div className={css.head}>
        <h5>{title}</h5>
        <CloseBtn close={close}/>
    </div>;

const CloseBtn = ({close}:CloseBtnProps) =>
    <div onClick={close} className={css.closeBtn}>
        <div className={css.closeBtn_img}/>
    </div>;

const createOnWrapperClick = (close:Close) =>
    ({target}) => {
        const isMain = target.classList.contains(css.main);

        if (isMain) close();
    };