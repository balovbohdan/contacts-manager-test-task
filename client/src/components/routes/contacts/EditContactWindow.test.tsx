import * as React from 'react';
import {mount, shallow} from 'enzyme';

import {EditContactWindow} from './EditContactWindow';
import * as css from '../../modal-window/ModalWindow.css';

describe('EditContactWindow, root version', () => {
    const propsDef = {
        active: 1,
        close: jest.fn(),
        edit: jest.fn(),
        contacts: [{
            id: 1,
            name: 'name',
            img: '/user.svg',
            phone: '+380967164354'
        }]
    };

    it('Should match snapshot when active', () => {
        const props = { ...propsDef };
        const window = shallow(<EditContactWindow {...props}/>);

        expect(window).toMatchSnapshot();
    });

    it('Should match snapshot when not active', () => {
        const props = {
            ...propsDef,
            active: false
        };

        const window = shallow(<EditContactWindow {...props}/>);

        expect(window).toMatchSnapshot();
    });

    it('Should react on close btn click', () => {
        const props = {
            ...propsDef,
            close: jest.fn()
        };

        const window = mount(<EditContactWindow {...props}/>);

        window.find(`.${css.closeBtn}`).simulate('click');
        expect(props.close).toBeCalledTimes(1);
    });

    it('Should close after empty space click', () => {
        const props = {
            ...propsDef,
            close: jest.fn()
        };

        const window = mount(<EditContactWindow {...props}/>);

        window.find(`.${css.main}`).at(0).simulate('click');
        expect(props.close).toBeCalledTimes(1);
    });
});