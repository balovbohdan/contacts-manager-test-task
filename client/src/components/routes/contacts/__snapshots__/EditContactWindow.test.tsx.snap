// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditContactWindow, root version Should match snapshot when active 1`] = `
<Component
  close={[MockFunction]}
  contact={
    Object {
      "id": 1,
      "img": "/user.svg",
      "name": "name",
      "phone": "+380967164354",
    }
  }
  edit={[MockFunction]}
/>
`;

exports[`EditContactWindow, root version Should match snapshot when not active 1`] = `""`;
